### PR TITLE
update foreign-types to 0.5.0

### PR DESCRIPTION
**Pull request by wongsyrone**
Signed-off-by: Syrone Wong <wong.syrone@gmail.com>

### CODE REVIEW

File: openssl/src/macros.rs [hunk 1]
```diff
@@ -144,20 +144,16 @@ macro_rules! foreign_type_and_impl_send_sync {
...
                 $(#[$borrowed_attr])*
-                pub struct $borrowed;
-            }
+                pub unsafe type $owned : Sync + Send {
 
-            unsafe impl Send for $owned{}
-            unsafe impl Send for $borrowed{}
-            unsafe impl Sync for $owned{}
-            unsafe impl Sync for $borrowed{}
+                    $(#[$impl_attr])*
```
Comment (wongsyrone) [inline openssl/src/macros.rs:151]: I don't know whether this should be put at `pub unsafe type $owned : Sync + Send {`

**Comment by wongsyrone** [comment]
This is a cherry-pick from my own branch, CI passed: https://github.com/wongsyrone/rust-openssl/runs/2532690133

**Comment by sfackler** [comment]
As mentioned on that issue, this is a breaking change.

**Comment by wongsyrone** [comment]
I know it becomes unsafe trait

**Comment by sfackler** [comment]
So we aren't going to be able to make this change until the next major version bump of this crate, which is not for a while.

**Comment by wongsyrone** [comment]
When will the next major version possibly be released? OpenSSL 3.0 stable?

**Comment by sfackler** [comment]
After the 3.0 release, probably some time late this year.

**Comment by wongsyrone** [comment]
LGTM.

**Comment by wongsyrone** [comment]
@sfackler Any update on this? I see OpenSSL 3.0 support is merged.

**Pull request closed without merging**